Mathlib/Data/Option/Basic.lean

```diff
@@ -88,15 +88,18 @@ theorem bind_eq_some' {x : Option α} {f : α → Option β} {b : β} :
     x.bind f = some b ↔ ∃ a, x = some a ∧ f a = some b := by
   cases x <;> simp
 
-theorem bind_congr {f g : α → Option β} {x : Option α}
-    (h : ∀ a ∈ x, f a = g a) : x.bind f = x.bind g := by
-  cases x <;> simp only [some_bind, none_bind, mem_def, h]
-
 @[congr]
 theorem bind_congr' {f g : α → Option β} {x y : Option α} (hx : x = y)
     (hf : ∀ a ∈ y, f a = g a) : x.bind f = y.bind g :=
   hx.symm ▸ bind_congr hf
 
+@[deprecated bind_congr (since := "2025-03-20")]
```

Comment:
I don't understand. If there is now `bind_congr` in core, do we want to give users a warning when they use it? Maybe I don't understand how `deprecated` works.

Reply:
the lemma got moved up to Lean, but Paul also changed the statement slightly.

So I left a copy in Mathlib, with an additional prime on the name, but deprecated it. It's not ideal, but leaves slightly more of a breadcrumb if anyone goes looking. It didn't break anything in Mathlib, so hopefully no one even notices.